src/main/java/io/cryostat/discovery/CustomDiscovery.java

```diff
@@ -128,13 +128,14 @@ public Response create(
         return doV2Create(target, Optional.ofNullable(credential), dryrun, storeCredentials);
     }
 
-    @Transactional
-    @Blocking
     Response doV2Create(
             Target target,
             Optional<Credential> credential,
             boolean dryrun,
             boolean storeCredentials) {
+        if (!QuarkusTransaction.isActive()) {
+            QuarkusTransaction.begin();
```

Comment:
I guess it would commit anyway because the callers are using `@Transactional`, and those ones are actually proper managed uses of the annotation, but it's better to be clear and explicit about the rollback/commit behaviour in this helper.